core/src/Streamly/Internal/Data/ParserK/Type.hs

```diff
@@ -487,17 +491,24 @@ adaptWith
     -> Input a
     -> m (Step a m r)
 adaptWith pstep initial extract cont !relPos !usedCount !input = do
+    -- We ensure that this does not happen
+#ifdef DEBUG
+    when (relPos /= 0) $ error $ "adaptWith: entry relPos " ++ show relPos
```

Comment:
I did not understand this statement.
Should `relPos` always be `0` here?

Reply:
We think that it should be 0, but we are not taking the risk of being wrong. The code is written to handle even if it is not 0. The debug statement will tell us if the assumption is wrong.